### PR TITLE
TC-780 Add top banners

### PR DESCRIPTION
**Pull request by nickbrowne**
Adds two new components for the donation campaign:

## DonationBanner
Screenshot at 2019-04-24 13-31-46

## ThinDonationBanner
Screenshot at 2019-04-24 13-31-57

### CODE REVIEW

**Comment by nickbrowne** [comment]
You may notice that the "x" icon is slightly further in from the designs on zeplin, this matches the position in the `DonationDialog`, and is to give the hover/active animations some breathing room. Has Zoe's approval.

File: src/promo/DonationBanner.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  onClose: PropTypes.func,
+
+  /**
+   * Accessibility text for close button
+   */
+  closeText: PropTypes.string
+}
+
+DonationBanner.defaultProps = {
+  href: 'http://example.com',
```
Comment (jameshill) [inline src/promo/DonationBanner.jsx]: Should we just make the default https://donate.theconversation.com?

Comment (jameshill) [reply src/promo/DonationBanner.jsx]: We didn't make it the default on the `EndOfArticle`, but just thought going forward perhaps we should.

**Comment by jameshill** [comment]
Given that the default is also in `EndOfArticle` i'm voting for changing it in a separate PR.

**Comment by nickbrowne** [comment]
Yeah it's currently `'http://example.com'` for `DonationDialog` as well.

**Pull request merged**